### DISCUSSION TITLE
Clubs and Events Design

### DISCUSSION THREAD

**Discussion by Samy-33** [BitByte-TPC/gymkhana#169] [Ideas]
[Clubs and Events Design Doc](https://github.com/BitByte-TPC/gymkhana/wiki/Clubs-and-events-Design-Doc)
- Problems with using hooks
  - Application deployed on the Intranet won't be accessible from the Internet. As a result hooks won't work.
  - It complicates the implementation unnecessarily. We could store a link to google form if a club wants to get more information. But important and common information could be accepted in event registration form in the application itself.

> Each club will have an event reports page which will be visible to college administrators and will be editable by the club's core members. This will contain event reports(Google docs) for all events conducted by the club. The link to this page will be visible on the sidebar.

Why are we using `Google docs` for the report?

> RSVP

Let's explain what this means in glossary or add it's full form with it's first occurrence: `We will also show the number of RSVPs (R... S... V... P...) for the past and ongoing events.`

**Comment by Aksh-Bansal-dev** [comment]
> Application deployed on the Intranet won't be accessible from the Internet. As a result hooks won't work.

Yeah. I forgot this will be on intranet.

> Why are we using Google docs for the report?

I'm not sure about other clubs, but in TPC, we currently use Google docs for all reports. And if we don't use google docs and use some npm package for the editor, it will not be as good(feature-rich). Also, there may be some formatting issues if we copy-paste content from docs to 'our editor'.

**Reply by Samy-33** [reply]
Does the reports page contain only a link to Google Doc? or some other information?

**Reply by Aksh-Bansal-dev** [reply]
Reports for all events(each event will have a separate report). We may also add a filter or pagination or something to view reports for a specific session (2021-2022).

**Comment by Samy-33** [comment]
I am not aware of these reports. So sorry for stupid questions.

- Who prepares these reports?
- Have we seen any reports till now, can you provide an example report?
- Is it necessary to provide these reports for events?

**Reply by Aksh-Bansal-dev** [reply]
> Who prepares these reports?

Core members.

> Have we seen any reports till now, can you provide an example report?

I will share it with you on discord.

> Is it necessary to provide these reports for events?

Yes.

**Reply by Samy-33** [reply]
Okay, let's have an option to store a link to this doc along with some trivial analysis on the event like # of registrations, attendees, event rating etc.
